### PR TITLE
[Live] coerce falsey strings to scalar types

### LINKED ISSUE #526: [Live] scalar type coercion when hydrating
If a component property's type-hint is `int` and you try to hydrate with `''`, you get an error. I propose we automatically cast simple, scalar types:

- `int`: `''` or `'  '` => `0`
- `float`: `''` or `'  '` => `0.0`
- `bool`:  `''` or `'  '` => `false`
- `?int`: `''` or `'  '` => `null`
- `?float`: `''` or `'  '` => `null`
- `?bool`:  `''` or `'  '` => `null`

How to handle?
- `int`: `'apple'`?

### PR DESCRIPTION
**Pull request by kbond**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| Tickets       | Fix #526
| License       | MIT

I chose to allow _weird_ values to be cast to the proper type instead of throwing an exception (ie `0 === (int) 'apple'`).

### CODE REVIEW

File: src/LiveComponent/src/LiveComponentHydrator.php [hunk 1]
```diff
@@ -148,6 +148,8 @@ public function hydrate(object $component, array $data, string $componentName):
             if ($method = $liveProp->hydrateMethod()) {
                 // TODO: Error checking
                 $value = $component->$method($value);
+            } elseif (\is_string($value) && $type && \in_array($type->getName(), ['int', 'float', 'bool'], true)) {
+                $value = self::coerceScalarValue($value, $type);
             } elseif (!$value && $type && $type->allowsNull() && is_a($type->getName(), \BackedEnum::class, true) && !\in_array($value, array_map(fn (\BackedEnum $e) => $e->value, $type->getName()::cases()))) {
                 $value = null;
             } elseif (null !== $value && $type && !$type->isBuiltin()) {
```
Comment (weaverryan) [inline src/LiveComponent/src/LiveComponentHydrator.php:155]: This area is increasingly feeling "ad-hoc", but it's well-covered in tests. And if there is some broader solution for handling these edge cases, it'll become obvious eventually 👍

**Comment by weaverryan** [comment]
Thanks Kevin!
